sw/device/tests/crypto/cryptotest/firmware/ibex_sca.c

```diff
@@ -10,25 +10,253 @@
 #include "sw/device/lib/testing/test_framework/ottf_test_config.h"
 #include "sw/device/lib/testing/test_framework/ujson_ottf.h"
 #include "sw/device/lib/ujson/ujson.h"
+#include "sw/device/sca/lib/prng.h"
 #include "sw/device/sca/lib/sca.h"
 #include "sw/device/tests/crypto/cryptotest/firmware/sca_lib.h"
 #include "sw/device/tests/crypto/cryptotest/firmware/status.h"
 #include "sw/device/tests/crypto/cryptotest/json/ibex_sca_commands.h"
 
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
 
+// NOP macros.
+#define NOP1 "addi x0, x0, 0\n"
+#define NOP10 NOP1 NOP1 NOP1 NOP1 NOP1 NOP1 NOP1 NOP1 NOP1 NOP1
+#define NOP100 NOP10 NOP10 NOP10 NOP10 NOP10 NOP10 NOP10 NOP10 NOP10 NOP10
+
 // Buffer to allow the compiler to allocate a safe area in Main SRAM where
 // we can do the write/read test without the risk of clobbering data
 // used by the program.
 OT_SECTION(".data")
 static volatile uint32_t sram_main_buffer[8];
+static volatile uint32_t sram_main_buffer_batch[256];
 
 /**
- * ibex.sca.tl_write
+ * ibex.sca.tl_write_batch_fvsr_fix_address
  *
  * This SCA penetration test executes the following instructions:
+ * - Generate num_iteration FvsR values using the software LFSR.
  * - Loop num_iterations:
  *  - Set trigger
+ *  - Write data over TL-UL into a single position in SRAM.
+ *  - Unset trigger
+ *
+ * SCA traces are captured during trigger_high & trigger_low.
+ *
+ * @param uj The received uJSON data.
+ */
+status_t handle_ibex_sca_tl_write_batch_fvsr_fix_address(ujson_t *uj) {
+  // Get number of iterations and fixed data.
+  ibex_sca_test_fvsr_t uj_data;
+  TRY(ujson_deserialize_ibex_sca_test_fvsr_t(uj, &uj_data));
+  if (uj_data.num_iterations >= 256) {
```

Comment:
Could `TRY_CHECK()` work for you here?
https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/test_framework/check.h;drc=80ecd2fc15d24d807610e37d4e860efe3d3c0ff5;l=79
Or do you specifically want to return the `ABORTED()` status?

Reply:
Good point, I was not aware of this function. Replaced the manual checks with `TRY_CHECK()` as suggested.